File: NewsDemo/app/src/test/java/com/kevin/newsdemo/utils/ViewUtilsTest.java
package com.kevin.newsdemo.utils;

import android.text.SpannableStringBuilder;
import android.widget.EditText;
import androidx.test.ext.junit.runners.AndroidJUnit4;
import org.junit.Assert;
import org.junit.Test;
import org.junit.runner.RunWith;

import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;

/**
 * Created by kevin on 2019/07/21 13:44.
 */
@RunWith(AndroidJUnit4.class)
public class ViewUtilsTest {

    @Test
    public void should_getText() {
        EditText editText = mock(EditText.class);

        final String expected = "123";
        when(editText.getText()).thenReturn(new SpannableStringBuilder(expected));
        String out = ViewUtils.getText(editText);
        Assert.assertEquals(expected, out);
    }
}

File: NewsDemo/app/src/test/java/com/kevin/newsdemo/utils/ViewUtilsTest.kt
package com.kevin.newsdemo.utils

import android.text.SpannableStringBuilder
import android.widget.EditText
import androidx.test.ext.junit.runners.AndroidJUnit4
import org.junit.Assert
import org.junit.Test
import org.junit.runner.RunWith
import org.mockito.Mockito.`when`
import org.mockito.Mockito.mock

/**
 * Created by kevin on 2019/07/21 13:44.
 */
@RunWith(AndroidJUnit4::class)
class ViewUtilsTest {

    @Test
    fun should_getText() {
        val editText = mock(EditText::class.java)

        val expected = "123"
        `when`(editText.text).thenReturn(SpannableStringBuilder(expected))
        val out = ViewUtils.getText(editText)
        Assert.assertEquals(expected, out)
    }
}
